src/main/java/it/gov/pagopa/pu/debtpositions/service/update/DebtPositionCancelInstallmentServiceImpl.java

```diff
@@ -0,0 +1,69 @@
+import it.gov.pagopa.pu.debtpositions.service.create.debtposition.DebtPositionProcessorService;
+import it.gov.pagopa.pu.debtpositions.service.statusalign.DebtPositionHierarchyStatusAlignerService;
+import it.gov.pagopa.pu.debtpositions.service.sync.DebtPositionSyncService;
+import it.gov.pagopa.pu.organization.dto.generated.Organization;
+import it.gov.pagopa.pu.workflowhub.dto.generated.PaymentEventType;
+import lombok.extern.slf4j.Slf4j;
+
+import org.apache.commons.lang3.tuple.Pair;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+@Slf4j
+public class DebtPositionCancelInstallmentServiceImpl extends BaseDebtPositionOperationService implements DebtPositionCancelInstallmentService {
+
+  protected DebtPositionCancelInstallmentServiceImpl(AuthorizeOperatorOnDebtPositionTypeService authorizeOperatorOnDebtPositionTypeService,
+                                                     DebtPositionService debtPositionService,
+                                                     DebtPositionSyncService debtPositionSyncService,
+                                                     DebtPositionProcessorService debtPositionProcessorService,
+                                                     OrganizationService organizationService,
+                                                     DebtPositionMapper debtPositionMapper,
+                                                     DebtPositionHierarchyStatusAlignerService debtPositionHierarchyStatusAlignerService) {
+    super(authorizeOperatorOnDebtPositionTypeService, debtPositionService, debtPositionSyncService, debtPositionProcessorService, organizationService, debtPositionMapper, debtPositionHierarchyStatusAlignerService);
+  }
+
+  @Override
+  public Pair<DebtPositionDTO, String> cancelInstallment(DebtPositionDTO debtPositionDTO, List<InstallmentDTO> installments2operate, Boolean massive, String accessToken, String operatorExternalUserId) {
+    if(log.isDebugEnabled()) {
+      List<Long> installmentIds = installments2operate.stream().map(InstallmentDTO::getInstallmentId).toList();
+      log.debug("Cancelling installments with ids {} for debt position with id {}", installmentIds, debtPositionDTO.getDebtPositionId());
+    }
+
+    Pair<DebtPositionDTO, String> debtPositionUpdated = execute(debtPositionDTO, installments2operate, massive, PaymentEventType.DPI_CANCELLED, accessToken, operatorExternalUserId);
+
+    log.debug("Cancelled installments for debt position with id {}", debtPositionDTO.getDebtPositionId());
+    return debtPositionUpdated;
+  }
+
+  @Override
+  public DebtPositionDTO applyOperation(DebtPositionDTO debtPositionDTO, List<InstallmentDTO> installments2operate, String accessToken, Organization org) {
+    List<Long> installmentIds = installments2operate.stream().map(InstallmentDTO::getInstallmentId).toList();
+
+    if(log.isDebugEnabled()){
+      log.debug("Updating status cancelled for installments with ids {}", installmentIds);
+    }
```

Comment:
I was thinking that the installmentIds was used just inside the log message, the isLogDebugEnabled has not more usage (i write it in order to avoid to perform the stream operation
```suggestion
    List<Long> installmentIds = installments2operate.stream().map(InstallmentDTO::getInstallmentId).toList();

    log.debug("Updating status cancelled for installments with ids {}", installmentIds);
```